crates/mzoon/src/wasm_opt.rs

```diff
@@ -104,16 +105,20 @@ fn unpack_wasm_opt(tar_gz: Vec<u8>) {
     for entry in archive.entries()? {
         let mut entry = entry?;
         let path = entry.path()?;
-        let file_stem = path
-            .file_stem()
+        let file_name = path
+            .file_name()
             .ok_or(anyhow!("Entry without a file name"))?;
-        if file_stem != "wasm-opt" {
-            continue;
-        }
-        let mut destination = PathBuf::from("frontend");
-        destination.push(path.file_name().unwrap());
+
+        let destination = match file_name.to_str() {
+            Some("wasm-opt")          => PathBuf::from("frontend/binaryen/bin/wasm-opt"),
+            Some("libbinaryen.dylib") => PathBuf::from("frontend/binaryen/lib/libbinaryen.dylib"),
+            _ => continue,
+        };
+        create_dir_all(destination.parent().unwrap())?;
         entry.unpack(destination)?;
-        return;
     }
-    Err(anyhow!("Failed to find wasm-opt in the downloaded archive"))?;
+
+    if !(PathBuf::from("frontend/binaryen/bin/wasm-opt").exists() && PathBuf::from("frontend/binaryen/lib/libbinaryen.dylib").exists()) {
+        Err(anyhow!("Failed to find wasm-opt in the downloaded archive"))?
+    };
```

Comment:
This would always fail on non-mac OS. Also I would use two different error messages for both files.

Perhaps I would replace these checks completely with one more reliable check - we can just call the `check_wasm_opt` function (it's in the same file) => if it's result is `Ok()` then `wasm-opt` has been installed correctly otherwise there is a problem and we can return a mapped error message like `wasm-opt installation failed: {error:#}`.

Reply:
I was stupid so it completely slipped my mind...😅

> we can just call the `check_wasm_opt` function

Your idea sounds nice. Let me give it a try!